File: java/src/main/java/codingdojo/AuctionMessageTranslator.java
package codingdojo;


import java.util.HashMap;

public class AuctionMessageTranslator {
    private final AuctionEventListener listener;

    public AuctionMessageTranslator(AuctionEventListener listener) {
        this.listener = listener;
    }

    public void processMessage(String message) {

        if (message.contains("CLOSE")) {
            // bug: should be listener.auctionClosed()
            listener.auctionFailed();
        } else if (message.contains("PRICE")) {
            var data = new HashMap<String, String>();
            for (var element : message.split(";")) {
                var pair = element.split(":");
                data.put(pair[0].trim(), pair[1].trim());
            }
            var currentPrice = Integer.parseInt(data.get("CurrentPrice"));
            var increment = Integer.parseInt(data.get("Increment"));
            var bidder = data.get("Bidder");

            // bug: should be listener.currentPrice(currentPrice)
            listener.auctionClosed();
        } else {
            listener.auctionFailed();
        }
    }
}

File: kotlin/src/main/kotlin/org/sammancoaching/AuctionMessageTranslator.kt
package org.sammancoaching


class AuctionMessageTranslator(val listener: AuctionEventListener) {

    fun processMessage(message: String) {
        if (message.contains("CLOSE")) {
            // bug: should notify listener
        } else if (message.contains("PRICE")) {
            val data = HashMap<String, String>()
            for (element in message.split(";")) {
                val pair = element.split(":")
                data[pair[0].trim()] = pair[1].trim()
            }
            val currentPrice = data.get("CurrentPrice")!!.toInt()
            val increment = data.get("Increment")!!.toInt()
            val bidder = data.get("Bidder")

            // bug: should notify listener
        } else {
            // bug: should notify listener
        }
    }
}